lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -301,6 +302,14 @@ sub done {
     # use $res as a result, it is recomputed result by scheduler
     $self->emit_event('openqa_job_done', {id => $jobid, result => $res, newbuild => $newbuild});
 
+    # notify workers if job has any chained children
+    my $children = $job->deps_hash->{children};
+    if (@{$children->{Chained}} && grep { $res eq $_ } OpenQA::Schema::Result::Jobs::OK_RESULTS) {
+        $self->app->log->debug("Job result OK and has chained children! Notifying workers");
+        my $ipc = OpenQA::IPC->ipc;
+        $ipc->scheduler('job_notify_workers');
```

Comment:
I'm for calling this directly - `OpenQA::Scheduler::Scheduler::job_notify_workers;` and later/or now move it to some sensible place.
See:
```perl
sub job_notify_workers {
    # notify workers about new job
    my $ipc = OpenQA::IPC->ipc;
    $ipc->websockets('ws_send_all', 'job_available');
}
```

so in this case WebUI->DBUS->Scheduler->DBUS->WebSockets and it waits for all.